src/Context.js

```diff
@@ -12,10 +14,25 @@ export const PathContext = createContext([])
 
 export const useConfirm = (shouldPrompt, message) => {
   return (callback) => {
-    if (shouldPrompt && !window.confirm(message)) {
-      return
+    const handleConfirmation = (result) => {
+      if (!result) {
+        return
+      }
+      callback()
+    }
+
+    if (shouldPrompt) {
+      var result = window.confirm(message)
+
+      // Some implementations (Thanks Tauri) return a promise
+      if (result instanceof Promise) {
+        result.then(handleConfirmation)
```

Comment:
If you make this an async function, then you can `await` the value whether or not it's a promise.

```
// You can await things that aren't promises.
true === await true;
true === await Promise.resolve(true);
```

Reply:
Thank you. As is very apparent, I only play a JS dev on TV

Improved in https://github.com/BlueWinds/bluescribe/pull/12/commits/71ec86b4ab0100e9062c519df2edaa9c7e209357